tensorboard/pip_package/pip_smoke_test.sh

```diff
@@ -0,0 +1,183 @@
+# limitations under the License.
+# ==============================================================================
+#
+# Smoke test for building, installing and basic usage of tensorboard pip package.
+#
+# Usage:
+#   pip_smoke_test.sh [--python3] [--retries <NUM_RETRIES>] [--port <PORT>]
+
+set -e
+
+die() {
+  echo "$1" 1>&2
+  exit 1
+}
+
+PY_VERSION=2
+TEST_PORT=6006
+NUM_RETRIES=20
+while true; do
+  if [[ -z "$1" ]]; then
+    break
+  fi
+
+  if [[ "$1" == "--python3" ]]; then
+    PY_VERSION=3
+  elif [[ "$1" == "--retries" ]]; then
+    NUM_RETRIES="$2"
+    shift
+  elif [[ "$1" == "--port" ]]; then
+    TEST_PORT="$2"
+    shift
+  else
+    die "ERROR: Unrecognized argument $1"
+  fi
+  shift
+done
+
+echo
+echo "=== Performing Smoke test of tensorboard PIP package ==="
+echo "Settings:"
+echo "  PY_VERSION=${PY_VERSION}"
+echo "  TEST_PORT=${TEST_PORT}"
+echo "  NUM_RETRIES=${NUM_RETRIES}"
+echo
+
+# If virtualenv is not installed, install it.
+if [[ -z $(which virtualenv) ]]; then
+  echo
+  echo "Installing virtualenv"
+  echo
+  pip install virtualenv
+fi
+
+if [[ ${PY_VERSION} == 2 ]]; then
+  pip install wheel
+else
+  pip3 install wheel
+fi
+
+PIP_TMP_DIR=$(mktemp -d --suffix _tensorboard)
+
+echo
+echo "Building tensorboard pip package in directory: ${PIP_TMP_DIR}"
+echo
+
+bazel build -c opt tensorboard/pip_package:build_pip_package
+bazel-bin/tensorboard/pip_package/build_pip_package "${PIP_TMP_DIR}"
+
+if [[ ${PY_VERSION} == 2 ]]; then
+  WHL_PATH=$(ls ${PIP_TMP_DIR}/tensorflow_tensorboard*-py2-*.whl)
+elif [[ ${PY_VERSION} == 3 ]]; then
+  WHL_PATH=$(ls ${PIP_TMP_DIR}/tensorflow_tensorboard*-py3-*.whl)
+fi
+
+if [[ -z ${WHL_PATH} ]]; then
+  die "ERROR: Cannot find the built pip package for python version ${PY_VERSION}"
+else
+  echo
+  echo "Found the built pip package for python version ${PY_VERSION} at:"
+  echo "  ${WHL_PATH}"
+  echo
+fi
+
+# Create virtualenv directory, cleanly (i.e., no --system-site-packages).
+VENV_TMP_DIR=$(mktemp -d --suffix _tensorboard_venv)
+
+echo
+echo "Creating virtualenv directory at: ${VENV_TMP_DIR}"
+echo
+
+if [[ ${PY_VERSION} == 2 ]]; then
+  PYTHON_PATH="/usr/bin/python"
+  TF_NIGHTLY_URL='https://ci.tensorflow.org/view/Nightly/job/nightly-matrix-cpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=cpu-slave/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp27-none-linux_x86_64.whl'
+elif [[ ${PY_VERSION} == 3 ]]; then
+  PYTHON_PATH="/usr/bin/python3"
+  TF_NIGHTLY_URL='https://ci.tensorflow.org/view/Nightly/job/nightly-matrix-cpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON3,label=cpu-slave/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp34-cp34m-linux_x86_64.whl'
+fi
+
+virtualenv -p ${PYTHON_PATH} "${VENV_TMP_DIR}"
+source ${VENV_TMP_DIR}/bin/activate
+
+# Install the dependency, tensorflow, first.
+echo
+echo "Installing nightly tensorflow pip package."
+echo
+
+pip install ${TF_NIGHTLY_URL}
```

Comment:
Missed this one :-) (shellcheck decided not to complain because it can statically determine that `TF_NIGHTLY_URL` will not change under globbing and word splitting, but Google-style and I still prefer that you quote it.)